pkg/ccl/backupccl/show.go

```diff
@@ -38,6 +41,28 @@ import (
 	"github.com/cockroachdb/errors"
 )
 
+func checkShowBackupURIPrivileges(ctx context.Context, p sql.PlanHookState, uri string) error {
+	// Check if the user issuing the SHOW BACKUP needs to be of the admin role
+	// depending on the URI destination.
+	hasAdmin, err := p.HasAdminRole(ctx)
```

Comment:
nit: I'd invert this and only check if isAdmin if we care, i.e. if the URI says we need to.

Reply:
done.